### PR TITLE
Fix logging issues, cleanup config parsing

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightningnetwork/lnd/issues/5936.
Fixes https://github.com/lightningnetwork/lnd/issues/5927.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -685,12 +716,16 @@ func ValidateConfig(cfg Config, usageMessage string,
...
+			cfg.Watchtower.TowerDir = filepath.Join(
+				cfg.DataDir, defaultTowerSubDirname,
+			)
 		}
 	}
 
-	funcName := "loadConfig"
+	funcName := "ValidateConfig"
+	mkErr := func(format string, args ...interface{}) error {
+		return fmt.Errorf(funcName+": "+format, args...)
```
Comment (yyforyongyu) [inline config.go:727]: Should the `mkErr` return a `usageError` to get it printed to console?

File: config.go [hunk 2]
```diff
@@ -727,6 +727,10 @@ func ValidateConfig(cfg Config, interceptor signal.Interceptor, fileParser,
 		return fmt.Errorf(funcName+": "+format, args...)
 	}
 	makeDirectory := func(dir string) error {
+		if directoryExists(dir) {
+			return nil
+		}
+
```
Comment (yyforyongyu) [inline config.go:733]: I think `MkdirAll` will do nothing if the dir doesn't exist?

File: lnd.go [hunk 3]
```diff
@@ -341,8 +343,7 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 			cfg.Cluster.ID)
 
 		if err := leaderElector.Campaign(electionCtx); err != nil {
-			ltndLog.Errorf("Leadership campaign failed: %v", err)
-			return err
+			return mkErr("leadership campaign failed: %v", err)
```
Comment (yyforyongyu) [inline lnd.go:346]: 👍 nice about the  lowercase `l`

File: lnd.go [hunk 4]
```diff
@@ -549,22 +535,18 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
...
+		return mkErr("unable to start RPC server: %v", err)
 	}
 	defer rpcServer.Stop()
 
 	// We transition the RPC state to Active, as the RPC server is up.
 	interceptorChain.SetRPCActive()
 
 	if err := interceptor.Notifier.NotifyReady(true); err != nil {
-		return err
+		return mkErr("error notifying rady: %v", err)
```
Comment (yyforyongyu) [inline lnd.go:549]: typo: rady -> ready

**Review by yyforyongyu — COMMENTED** [review]
Cool. Love that all the long lines are fixed! Only got a few questions in the comments.

File: config.go [hunk 5]
```diff
@@ -1063,36 +1082,39 @@ func ValidateConfig(cfg Config, usageMessage string,
...
-		cfg.Litecoin.ChainDir = filepath.Join(cfg.DataDir,
-			defaultChainSubDirname,
-			chainreg.LitecoinChain.String())
+		cfg.Litecoin.ChainDir = filepath.Join(
+			cfg.DataDir, defaultChainSubDirname,
+			chainreg.LitecoinChain.String(),
+		)
 
-		// Finally we'll register the litecoin chain as our current
+		// Finally, we'll register the litecoin chain as our current
```
[the remark this answers is not in the document]
Comment (guggero) [reply config.go:1121]: My IDE now also highlights potential grammar style fixes. My OCD is getting rid of the IDE hints, but maybe I should just turn those off :joy:

File: config.go (hunk 1 above)
Comment (guggero) [reply config.go:727]: No, by default we don't want the usage string (e.g. `Use lnd -h to show usage`) to be printed, only when we know the user provided an incorrect value.

File: config.go (hunk 2 above)
Comment (guggero) [reply config.go:733]: Hmm... You are right. So I think https://github.com/lightningnetwork/lnd/issues/5927 is caused by incorrect permissions on the directory in the first place.

Comment (guggero) [reply config.go:733]: I dropped the commit.

Comment (guggero) [reply config.go]: Actually fixed the problem this time, see 875d6553.

File: config.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (yyforyongyu) [reply config.go:1099]: I really should have one of those gramma checking plugins too haha

**Review by Roasbeef — APPROVED** [review]
LGTM 🪡

**Pull request merged**